### DISCUSSION TITLE
Rusqlite Driver supported

### DISCUSSION THREAD

**Discussion by realme0213** [SeaQL/sea-orm#721] [Q&A] [answered]
## Summary
Hi, thank you for your great work! `sea-orm` built upon `sea-query`, and I find that `sea-query` already supports `rusqlite` with feature flag, is it possible to support `rusqlite` by `sea-orm`.

## Motivation
I want to build a cross platform app with bundled sqlite

**Comment by tyt2y3** [comment]
I don't think we are going to support rusqlite officially in near future. Fundamentally it does not provide an async and so does not quite fit well with SeaORM.

**Comment by arihant2math** [comment]
@tyt2y3 This is an old discussion but I'm trying to use sea-orm for an indexeddb implementation in [servo](https://servo.org). sqlite is synchronous, so rusqlite should be a better/more performant backend than sqlx.

**Reply by tyt2y3** [reply]
performance aside, currently the SeaORM api is async. how would you imagine what the API looks like?
btw, your link above is not found.

**Reply by arihant2math** [reply]
> performance aside, currently the SeaORM api is async. how would you imagine what the API looks like?

Likely something like `execute_sync` as a replacement for `execute` and similar sync counterparts for async functions/methods.
> btw, your link above is not found.

Markdown was interpreting it as a relative link, nothing important there however.

**Comment by tyt2y3** [comment, marked as answer]
Latest update:

[`sea-orm-sync`](https://crates.io/crates/sea-orm-sync) provides the full SeaORM API without requiring an async runtime, making it ideal for lightweight CLI programs with SQLite.

See the [quickstart example](https://github.com/SeaQL/sea-orm/blob/master/sea-orm-sync/examples/quickstart/src/main.rs) for usage.

See also https://www.sea-ql.org/blog/2025-12-12-sea-orm-2.0/

feel free to try it out and let me know
